### PR TITLE
Azure MachineSet: VM ScaleSet or AvailabilitySet

### PR DESCRIPTION
**Pull request by dkistner**
**What this PR does / why we need it**:
Add support for VirtualMachineScaleSet Orchestration Mode VM (VMO) in the `AzureMachineClass`.

Introduce a generic MachineSet configuration field in `AzureMachineClass`.
The `machineSet` field can be used to configure either a VMO (VirtualMachineScaleSet Orchestration Mode VM) or an AvailabilitySet for a MachineClass. Or whatever else will come in the future :)

The `availabilitySets` field will be kept for a while but is now deprecated in favour of the `machineSet` field.
The Azure compute sdk is also updated to support VMO.

**Release note**:

```improvement user
The machine-controller-manager supports now machines attached to Azure VirtualMachineScaleSet Orchestration Mode VM (VMO).
```

```improvement developer
The field `availabilitySets` in the `AzureMachineClass` is now deprecated in favour of the field `machineSet`, which allow to configure AvailabilitySets and VirtualMachineScaleSet Orchestration Mode VM (VMO). The field will be removed in the future.
```

### CODE REVIEW

**Comment by hardikdr** [comment]
Thanks for the PR @dkistner, we are kind of stuck at a couple of other stuff, will take a look soon.

**Review by AxiomSamarth — APPROVED** [review]

**Comment by AxiomSamarth** [comment]
@prashanth26  

I went through the code and based on the code comments, I could read and comprehend the code. I do not have any comments as such. However, I understood from the PR that the spec of `AzureMachineClass` has been introduced with a new field `machineSet` that can be used to configure the `VMO` or the `availabilitySet`. This will help me to implement my Azure OOT accordingly. Hope the intention to ask me to review this was for the same.

**Comment by dkistner** [comment]
@AxiomSamarth Yes basically I added the `machineset` field to have a simple interface for any kind of machineset which will come. We need to add support for VMOs now, but we will still have the availabilitysets. I just deprecated the old field for only availability sets.

**Review by prashanth26 — APPROVED** [review]
/lgtm

**Review by hardikdr — COMMENTED** [review]
Thanks for the PR @dkistner , just a minor comment, looks great otherwise.

**Review by MSSedusch — APPROVED** [review]

File: pkg/apis/machine/types.go [hunk 1]
```diff
@@ -984,6 +985,19 @@ type AzureSubnetInfo struct {
...
+	ID   string
+	Kind string
+}
+
+const (
+	// MachineSetKindAvailabilitySet is the machine set kind for AvailabilitySet
+	MachineSetKindAvailabilitySet string = "availabilityset"
+	// MachineSetKindVMO is the machine set kind for VirtualMachineScaleSet Orchestration Mode VM (VMO)
+	MachineSetKindVMO string = "vmo"
+)
```
[the remark this answers is not in the document]
Comment (dkistner) [reply pkg/apis/machine/types.go:999]: done :)

**Pull request merged**